airbyte-integrations/connectors/destination-snowflake-cortex/destination_snowflake_cortex/cortex_processor.py

```diff
@@ -0,0 +1,406 @@
+        catalog_provider: CatalogProvider,
+        temp_dir: Path,
+        temp_file_cleanup: bool = True,
+    ) -> None:
+        """Initialize the Snowflake processor."""
+        self.splitter_config = splitter_config
+        self.embedder_config = embedder_config
+        super().__init__(
+            sql_config=sql_config,
+            catalog_provider=catalog_provider,
+            temp_dir=temp_dir,
+            temp_file_cleanup=temp_file_cleanup,
+        )
+
+    def _get_sql_column_definitions(
+        self,
+        stream_name: str,
+    ) -> dict[str, sqlalchemy.types.TypeEngine]:
+        """Return the column definitions for the given stream.
+
+        Return the static static column definitions for cortex streams.
+        """
+        return {
+            DOCUMENT_ID_COLUMN: self.type_converter_class.get_string_type(),
+            CHUNK_ID_COLUMN: self.type_converter_class.get_string_type(),
+            METADATA_COLUMN: self.type_converter_class.get_json_type(),
+            DOCUMENT_CONTENT_COLUMN: self.type_converter_class.get_json_type(),
+            EMBEDDING_COLUMN: f"VECTOR(FLOAT, {self.embedding_dimensions})",
+        }
+
+    @overrides
+    def _write_files_to_new_table(
+        self,
+        files: list[Path],
+        stream_name: str,
+        batch_id: str,
+    ) -> str:
+        """Write files to a new table.
+
+        This is the same as PyAirbyte's SnowflakeSqlProcessor implementation, migrated here for
+        stability. The main differences lie within `_get_sql_column_definitions()`, whose logic is
+        abstracted out of this method.
+        """
+        temp_table_name = self._create_table_for_loading(
+            stream_name=stream_name,
+            batch_id=batch_id,
+        )
+        internal_sf_stage_name = f"@%{temp_table_name}"
+
+        def path_str(path: Path) -> str:
+            return str(path.absolute()).replace("\\", "\\\\")
+
+        for file_path in files:
+            query = f"PUT 'file://{path_str(file_path)}' {internal_sf_stage_name};"
+            self._execute_sql(query)
+
+        columns_list = [
+            self._quote_identifier(c)
+            for c in list(self._get_sql_column_definitions(stream_name).keys())
+        ]
+        files_list = ", ".join([f"'{f.name}'" for f in files])
+        columns_list_str: str = indent("\n, ".join(columns_list), " " * 12)
+
+        # following block is different from SnowflakeSqlProcessor
+        vector_suffix = f"::Vector(Float, {self.embedding_dimensions})"
+        variant_cols_str: str = ("\n" + " " * 21 + ", ").join([
+            f"$1:{self.normalizer.normalize(col)}{vector_suffix if 'embedding' in col else ''}"
+            for col in columns_list
+        ])
+        if self.sql_config.cortex_embedding_model:
+            # WARNING: This is untested and may not work as expected.
+            variant_cols_str += f"snowflake.cortex.embed('{self.sql_config.cortex_embedding_model}', $1:{DOCUMENT_CONTENT_COLUMN})"
```

Comment:
This is untested, but the idea would be to inject a SQL-based embedding here if native embedding is requested.